lib/internal/process/per_thread.js

```diff
@@ -293,55 +294,53 @@ function buildAllowedFlags() {
   // process.allowedNodeEnvironmentFlags.has() which lack leading dashes.
-  const nodeFlags = new SafeSet(ArrayPrototypeMap(allowedNodeEnvironmentFlags,
-                                                  trimLeadingDashes));
-
-  class NodeEnvironmentFlagsSet extends SafeSet {
-    constructor(...args) {
-      super(...args);
-
-      // The super constructor consumes `add`, but
-      // disallow any future adds.
-      ObjectDefineProperty(this, 'add', {
-        value: () => this
-      });
-    }
-
-    delete() {
-      // No-op, `Set` API compatible
-      return false;
-    }
-
-    clear() {
-      // No-op
-    }
-
-    has(key) {
-      // This will return `true` based on various possible
-      // permutations of a flag, including present/missing leading
-      // dash(es) and/or underscores-for-dashes.
-      // Strips any values after `=`, inclusive.
-      // TODO(addaleax): It might be more flexible to run the option parser
-      // on a dummy option set and see whether it rejects the argument or
-      // not.
-      if (typeof key === 'string') {
-        key = StringPrototypeReplace(key, replaceUnderscoresRegex, '-');
-        if (RegExpPrototypeTest(leadingDashesRegex, key)) {
-          key = StringPrototypeReplace(key, trailingValuesRegex, '');
-          return super.has(key);
-        }
-        return nodeFlags.has(key);
+  const nodeFlags =
+    ArrayPrototypeMap(allowedNodeEnvironmentFlags, trimLeadingDashes);
+
+  return new Proxy(ObjectFreeze(allowedNodeEnvironmentFlags), {
+    get(target, key, receiver) {
+      switch (key) {
+        case 'add':
+          // No-op, `Set` API compatible
+          return () => receiver;
+
+        case 'delete':
+          // No-op, `Set` API compatible
+          return () => false;
+
+        case 'clear':
+          // No-op, `Set` API compatible
+          return FunctionPrototype;
+
+        case 'has':
+        case 'includes':
+          return (key) => {
+            // This will return `true` based on various possible
+            // permutations of a flag, including present/missing leading
+            // dash(es) and/or underscores-for-dashes.
+            // Strips any values after `=`, inclusive.
+            // TODO(addaleax): It might be more flexible to run the option
+            // parser on a dummy option set and see whether it rejects the
+            // argument or not.
+            if (typeof key === 'string') {
+              key = StringPrototypeReplace(key, replaceUnderscoresRegex, '-');
+              if (RegExpPrototypeTest(leadingDashesRegex, key)) {
+                key = StringPrototypeReplace(key, trailingValuesRegex, '');
+                return ArrayPrototypeIncludes(target, key);
+              }
+              return ArrayPrototypeIncludes(nodeFlags, key);
+            }
+            return false;
+          };
+
+        case 'size':
+          return target.length;
+
+        default:
+          return ReflectGet(target, key);
```

Comment:
What happens if I Array.prototype.push.call on it?

Reply:
It throws with 

```
TypeError: Cannot define property 102, object is not extensible
    at Proxy.push (<anonymous>)
    at Object.<anonymous> (…/node/test/parallel/test-process-env-allowed-flags.js:73:26)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47
```

Let me add a test case for this.